sphinx/domains/python.py

```diff
@@ -1413,6 +1413,11 @@ def resolve_any_xref(self, env: BuildEnvironment, fromdocname: str, builder: Bui
 
         # always search in "refspecific" mode with the :any: role
         matches = self.find_obj(env, modname, clsname, target, None, 1)
+
+        if len(matches) > 1:
+            # Remove all duplicated matches
+            matches = [(name, obj) for name, obj in matches if not obj.aliased]
+
         for name, obj in matches:
             if obj[2] == 'module':
```

Comment:
Skips a list comprehension step
```suggestion

        for name, obj in matches:
            if len(matches) > 1 and obj.aliased:
                # Skip duplicated matches
                continue

            if obj[2] == 'module':
```

A

Reply:
Done (I've stored `len(matches) > 1` in a variable)